src/main/java/bookface/storage/JsonSerializableBookFace.java

```diff
@@ -79,6 +82,8 @@ public BookFace toModelType() throws IllegalValueException {
             Book book = jsonAdaptedBook.toModelType();
             if (bookFace.hasBook(book)) {
                 throw new IllegalValueException(MESSAGE_DUPLICATE_BOOK);
+            } else if (book.isLoaned() || book.getReturnDateString() != null) {
```

Comment:
To pass the test cases, you need to replace the null check with `!book.getReturnDateString().equals("")`

Reply:
Ah, silly mistake on my end. Thank you!